### PR TITLE
Intermittent Emissions Functionality

### PR DESCRIPTION
**Pull request by ThomasGalesloot**
# Pull Request Key Information

## Reason for change

Previously, the parameters indicating the intermittency of emissions were propagated down as source parameters, but were unused. Intermittency functionality for emissions was desired.

## What was changed

This change enables sources to create intermittent emissions and adds two new emission types: intermittent_repairable_emission and intermittent_non_repairable_emission. These emission types inherit from the repairable and non-repairable emission types, respectively, and implement intermittency functionality through a mixin class.

## Intended Purpose

Add ability to model intermittent emissions

## Level of version change required

N/A

## Testing Completed

All units tests pass:
[unit_test_results.txt](https://github.com/user-attachments/files/15745522/unit_test_results.txt)

New functionality is thoroughly unit tested. 

Performed a manual test using simple_test_case_1 parameter with the following modified virtual world and infrastructure files and evaluation single simulation results. Evaluation is also attached - see final 2 columns.

Test Inputs and evaluation: 
[Intermittency Test Inputs and Manual Evaluation.zip](https://github.com/user-attachments/files/15745530/Intermittency.Test.Inputs.and.Manual.Evaluation.zip)

## Target Issue

N/A

## Additional Information

Documentation will be incorporated as part of a different change.

### CODE REVIEW

File: LDAR_Sim/testing/unit_testing/test_virtual_world/test_intermittent_repairable_emission/test_activate.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+from testing.unit_testing.test_virtual_world.test_intermittent_repairable_emission.intermittent_repairable_emission_creation_fixtures import (  # noqa
+    intermittent_repairable_emission_creation_params_fixture,
+)
+from virtual_world.emission_types.intermittent_repairable_emission import (
+    IntermittentRepairableEmission,
+)
+
+
+def test_activate_correctly_sets_emitting_true_if_emission_activated(
```
Comment (SEAJang) [inline LDAR_Sim/testing/unit_testing/test_virtual_world/test_intermittent_repairable_emission/test_activate.py:32]: Nit: Should we not also test for if emission should not be activated?

File: LDAR_Sim/testing/unit_testing/test_virtual_world/test_intermittent_non_repairable_emission/test_activate.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+from testing.unit_testing.test_virtual_world.test_intermittent_non_repairable_emission.intermittent_non_repairable_emission_creation_fixtures import (  # noqa
+    intermittent_non_repairable_emission_creation_params_fixture,
+)
+from virtual_world.emission_types.intermittent_non_repairable_emission import (
+    IntermittentNonRepairableEmission,
+)
+
+
+def test_activate_correctly_sets_emitting_true_if_emission_activated(
```
Comment (SEAJang) [inline LDAR_Sim/testing/unit_testing/test_virtual_world/test_intermittent_non_repairable_emission/test_activate.py:32]: Nit: Should we not also test for if emission should not be activated?

File: LDAR_Sim/testing/unit_testing/test_virtual_world/test_sources/sources_testing_resources/mock_methods.py
Comment (SEAJang) [inline LDAR_Sim/testing/unit_testing/test_virtual_world/test_sources/sources_testing_resources/mock_methods.py:1]: Missing file docstring

**Review by SEAJang — APPROVED** [review]
Great job. 
A couple nit things to address, but otherwise the PR should be good to merge in. 
Regarding the documentation, could we that what is being written is in line with what was implemented?

File: LDAR_Sim/testing/unit_testing/test_virtual_world/test_intermittent_non_repairable_emission/test_activate.py (hunk 2 above)
Comment (ThomasGalesloot) [reply LDAR_Sim/testing/unit_testing/test_virtual_world/test_intermittent_non_repairable_emission/test_activate.py:32]: I'm only testing new behavior, most of activate functionality is inherited from the super, which is tested elsewhere.

File: LDAR_Sim/testing/unit_testing/test_virtual_world/test_intermittent_non_repairable_emission/intermittent_non_repairable_emission_creation_fixtures.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+------------------------------------------------------------------------------
+"""
+
+import pytest
+from datetime import date
+
+
+@pytest.fixture(name="intermittent_non_repairable_emission_creation_params")
+def intermittent_non_repairable_emission_creation_params_fixture() -> dict:
```
[the remark this answers is not in the document]
Comment (ThomasGalesloot) [reply LDAR_Sim/testing/unit_testing/test_virtual_world/test_intermittent_non_repairable_emission/intermittent_non_repairable_emission_creation_fixtures.py:27]: Probably a missed rename during copy pasting

**Pull request merged**